### PR TITLE
AM-927: Adding LD Bean caching.

### PR DESCRIPTION
**Pull request by nitishsingla1234**
### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/roleassignment/launchdarkly/LDFlagChecker.java [hunk 1]
```diff
@@ -5,44 +5,54 @@
...
 
-        boolean showFeature = client.boolVariation(flagName, user, false);
+        boolean showFeature = ldConfiguration.getLdClient().boolVariation(flagName, user, false);
 
-        client.close();
+        ldConfiguration.getLdClient().close();
         return showFeature;
     }
+
+    @Scheduled(fixedRate = 6000)
```
Comment (aashishaggarwal123) [inline src/main/java/uk/gov/hmcts/reform/roleassignment/launchdarkly/LDFlagChecker.java:50]: If it is 6 seconds, Isn't it too frequent. I think we can have at least a minute.

File: src/main/java/uk/gov/hmcts/reform/roleassignment/launchdarkly/LDFlagChecker.java [hunk 2]
```diff
@@ -5,44 +5,54 @@
...
+    @CacheEvict(allEntries = true, value = "ldconfig")
+    public void evictAllCaches() {
+        log.info("cleaning caches");
+        cacheManager.getCacheNames()
+                    .parallelStream()
```
Comment (aashishaggarwal123) [inline src/main/java/uk/gov/hmcts/reform/roleassignment/launchdarkly/LDFlagChecker.java:55]: Is parallel steam required here? there would be only one cache object for this implementation.

File: src/main/java/uk/gov/hmcts/reform/roleassignment/launchdarkly/LDConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    private String sdkProdKey;
+
+    @Bean
+    public LDClient getLdClient() {
+        String sdkKey;
+        if (environment.equalsIgnoreCase(Constants.AAT) || environment.equalsIgnoreCase(Constants.PROD)) {
+            sdkKey = sdkProdKey;
+        } else {
+            sdkKey = sdkTestKey;
+        }
```
Comment (aashishaggarwal123) [inline src/main/java/uk/gov/hmcts/reform/roleassignment/launchdarkly/LDConfiguration.java:32]: As we discussed last day, Shouldn't it be managed with single property which would carry different values injected as per test and prod envs.

File: src/main/java/uk/gov/hmcts/reform/roleassignment/launchdarkly/LDFlagChecker.java (hunk 1 above)
Comment (nitishsingla1234) [reply src/main/java/uk/gov/hmcts/reform/roleassignment/launchdarkly/LDFlagChecker.java]: We have removed caching and scheduling completely.

File: src/main/java/uk/gov/hmcts/reform/roleassignment/launchdarkly/LDFlagChecker.java (hunk 2 above)
Comment (nitishsingla1234) [reply src/main/java/uk/gov/hmcts/reform/roleassignment/launchdarkly/LDFlagChecker.java]: Removed the whole code. It's no longer required.

File: src/main/java/uk/gov/hmcts/reform/roleassignment/launchdarkly/LDConfiguration.java (hunk 3 above)
Comment (nitishsingla1234) [reply src/main/java/uk/gov/hmcts/reform/roleassignment/launchdarkly/LDConfiguration.java]: Removed the keys.

**Pull request merged**